sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -475,6 +475,16 @@ class SessionCatalog(
     externalCatalog.getTable(db, table)
   }
 
+  /**
+   * Retrieve the metadata of a catalog identifier.
+   */
+  def getTableMetadata(ident: Identifier): CatalogTable = {
```

Comment:
shall we turn `Identifier` into `TableIdentifier` and call the existing `getTableMetadata` method?

Reply:
Yeah, done in 305c316.